python/lsst/meas/deblender/baseline.cc

```diff
@@ -0,0 +1,116 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "lsst/afw/image/Image.h"
+#include "lsst/afw/image/MaskedImage.h"
+#include "lsst/afw/detection/Footprint.h"
+#include "lsst/afw/detection/Peak.h"
+
+#include "lsst/meas/deblender/Baseline.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace deblender {
+
+namespace{
+    template <typename ImagePixelT,
+              typename MaskPixelT=lsst::afw::image::MaskPixel,
+              typename VariancePixelT=lsst::afw::image::VariancePixel>
+    void declareBaselineUtils(py::module & mod, const std::string & suffix){
+        typedef typename lsst::afw::image::MaskedImage<ImagePixelT, MaskPixelT, VariancePixelT> MaskedImageT;
+        typedef typename PTR(lsst::afw::image::Image<ImagePixelT>) ImagePtrT;
+        typedef typename PTR(lsst::afw::detection::Footprint) FootprintPtrT;
+        
+        py::class_<BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>>
+            cls(mod, ("BaselineUtils"+suffix).c_str());
+        cls.def_static("symmetrizeFootprint",
+                       &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::symmetrizeFootprint);
+        // The function has a std::pair return value and a referenced boolean (patchedEdges) that is
+        // updated, so we combine them together to give a more pythonic interface to the function.
+        cls.def_static("buildSymmetricTemplate", [](MaskedImageT const& img,
+                                                    lsst::afw::detection::Footprint const& foot,
+                                                    lsst::afw::detection::PeakRecord const& pk,
+                                                    double sigma1,
+                                                    bool minZero,
+                                                    bool patchEdges){
+            bool patchedEdges;
+            std::pair<ImagePtrT, FootprintPtrT> result;
+            
+            result = BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::buildSymmetricTemplate(
+                img, foot, pk, sigma1, minZero, patchEdges, &patchedEdges);
+            return py::make_tuple(result.first, result.second, patchedEdges);
+        });
+        cls.def_static("medianFilter", &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::medianFilter);
+        cls.def_static("makeMonotonic",
+                       &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::makeMonotonic);
+        // The function has a std::pair return value and a referenced HeavyFootprint list (strays) that is
+        // updated, which must be returned to avoid using OPAQUE types.
```

Comment:
The problem is the c++ function takes a pointer to an empty vector as an input and adds elements to the vector. But when a python list is passed to the function the list is not populated. So instead we omit the vector/list as an input, create it in the lambda, and add the modified vector to the returns.